http-common/src/connector.rs

```diff
@@ -750,10 +737,22 @@ fn get_num_systemd_sockets() -> Result<Option<std::os::unix::io::RawFd>, String>
         None => return Ok(None),
     };
 
+    // If there is  no socket available, not match is possible.
     if listen_fds == 0 {
         return Ok(None);
     }
 
+    //If there is only one socket, we know this is the identity service which uses only one socket, so we have a match:
+    if listen_fds == 1 {
+        return Ok(Some(SD_LISTEN_FDS_START as i32));
+    }
+
+    //if there is more than 1 and we don't have a socket name to match, this is edged telling us that there is no systemd socket we can match.
+    let socket_name = match socket_name {
+        Some(socket_name) => socket_name,
+        None => return Ok(None),
+    };
+
     // fcntl(CLOEXEC) all the fds so that they aren't inherited by the child processes.
     // Note that we want to do this for all the fds, not just the one we're looking for.
     for fd in SD_LISTEN_FDS_START..(SD_LISTEN_FDS_START + listen_fds) {
```

Comment:
This needs to be done in any situation where there's an fd. So this needs to be done before the early return in the `listen_fds == 1` case. Just move it up before your line 745.

Reply:
Good catch. I dragged and drop some code and forgot about it.